### PR TITLE
Init v0.1.0

### PR DESCRIPTION
**Pull request by Parassharmaa**
- Add LambdaPackaging
- Add LambdaLayerPackaging
- Add Readme 
- Pip Packaging
- Add example

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Pululi
+pulumi==1.8.1
+pulumi_aws==1.17.0
+pulumi_docker
+requirements-parser
```
Comment (LouisAmon) [inline requirements.txt]: Nice find 👍

File: lambda_packaging/pip_requirements.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        """
+        self.generate_requirements_file()
+
+        self.pip_cmd.append(self.target_requirements_path)
+        self.pip_cmd.append('--upgrade')
+        self.pip_cmd.append(f'--target={self.install_path}')
+        if self.dockerize:
+            DockerizePip(self.project_root, self.runtime)
+        else:
+            subprocess.run(self.pip_cmd)
```
Comment (LouisAmon) [inline lambda_packaging/pip_requirements.py]: This module should just be a part of the `LambdaLayerPackage` component

File: lambda_packaging/docker.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                                 name=self.docker_image(runtime),
+                                 keep_locally=True)
+        
+        # docker cmd to run in the container
+        container_run_cmd = f'"cd {self.container_path}; pip install -r requirements.txt -t {path.join(self.install_folder)}"'
+        
+        # run container and install requirements
+        self.container = Container("lambda-python",
+                                   image=self.image.name,
+                                   command=["bash", "-c", container_run_cmd],
```
Comment (LouisAmon) [inline lambda_packaging/docker.py]: Why do you need the `bash -c` ?

Couldn't you split your `container_run_cmd ` string into a list of statements and use that as the `command` argument ?

File: lambda_packaging/docker.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import pulumi
+import pulumi_docker
+from os import path
+from pulumi_docker import RemoteImage, Container, Volume
+from .config import Config
+
+
+class DockerizePip(Config):
```
Comment (LouisAmon) [inline lambda_packaging/docker.py]: I don't necessarily mind having these classes... but subclassing a `Config` is definitely an antipattern!
The proper pattern would be to pass the config as input parameter but even still, I think that's overkill for what we're doing here

I think it's best that `DockerizePip` and `PipRequirements` become member functions of the `LambdaLayerPackage` component.

File: README.md [hunk 5]
```diff
@@ -1,62 +1,53 @@
...
-│   │       └── label.py
-│   ├── label.py
-│   ├── stacks
-│   │   └── Pulumi.dev.yaml
-│   ├── storage.py
-│   └── utils.py
-└── requirements
-    ├── dynamic_providers.txt
-    ├── global.txt
-    └── lambda.txt
```
Comment (LouisAmon) [inline README.md]: Could you please remake the directory structure ?

You can use `tree` in linux to generate it automatically

File: example/__main__.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+from lambda_packaging import LambdaPackaging, LambdaLayerPackaging
+
+lambda_package = LambdaPackaging(
+    'example-test', layer=True, exclude=['__main__.py'])
+
+lambda_layer = LambdaLayerPackaging('example-layer-test',
+                                    runtimes=['python3.6'],
+                                    layer_name="example-layer",
+                                    description="Example Layer for Testing",
+                                    exclude=[".gitignore", '__main__.py'])
```
[the remark this answers is not in the document]
Comment (LouisAmon) [reply example/__main__.py]: Btw there's a lib called `autopep8` that you can use to automatically apply code quality to the whole repo ;)

File: lambda_packaging/docker.py (hunk 3 above)
Comment (Parassharmaa) [reply lambda_packaging/docker.py]: The command looks like this:

```
bash -c "cd /io/; pip install -r requirements.txt -t ./requirements/"
```

I could not figure out the way to split it into the list!

File: lambda_packaging/docker.py (hunk 4 above)
Comment (Parassharmaa) [reply lambda_packaging/docker.py]: One major advantage of keeping classes separate is that these classes can be scaled easily with more features like reducing package size, slimming requirements, etc, in future versions as done in serverless-requirements.

Also, LambdaLayerPackage and LambdaPackage both use the ZipPackage module.

I have added docker class as a method of PipRequirements.

What do you say?

File: lambda_packaging/components.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        self.name = name
+        self.requirements_path = requirements_path
+        self.runtime = runtime
+        self.layer = layer
+        self.dockerize = dockerize
+        self.no_deploy = no_deploy
+        self.include = include
+        self.exclude = exclude
+        self.project_root = os.path.dirname(
+            os.path.abspath(sys.modules['__main__'].__file__))
```
[the remark this answers is not in the document]
Comment (Parassharmaa) [reply lambda_packaging/components.py]: Presently it tries to assume absolute path from __main__ file (that would __main__.py in pulumi).

Also, the use of this is to find requirements.txt. There is a parameter in LambdaPackage(requirements_path="./requirements.txt"). If one wants then can pass its path relative to __main__.py file.

File: example/requirements.txt [hunk 8]
```diff
@@ -0,0 +1 @@
+pylint
```
[the remark this answers is not in the document]
Comment (Parassharmaa) [reply example/requirements.txt]: I was using it for just testing the packaging of the pip requirements. I have updated it.

**Pull request merged**